src/local/ample.lean

```diff
@@ -13,9 +15,90 @@ vector spaces with their natural topology.
 
 open set
 
-variables {F : Type*} [add_comm_group F] [vector_space ℝ F] [topological_space F]
+variables {E : Type*} [add_comm_group E] [vector_space ℝ E] [topological_space E]
 
-/-- A subset of a topological real vector space is ample if the convex hull of each of its 
+/-- A subset of a topological real vector space is ample if the convex hull of each of its
 connected components is the full space. -/
-def ample_set (s : set F) := 
-∀ x : s, convex_hull (subtype.val '' (connected_component x)) = univ
\ No newline at end of file
+def ample_set (s : set E) :=
+∀ x : s, convex_hull (subtype.val '' (connected_component x)) = univ
+
+local notation ` I ` := interval (0 : ℝ) 1
+
+def path_connected {E : Type*} [topological_space E] {F : set E} (x y : F) : Prop :=
```

Comment:
Should this be `path_connected_within`?
In general, these declarations seem to make a lot of use of coercing sets to subtypes... not sure if that is a good idea.

Reply:
I think I borrowed this idea from the given definition of `ample` at the top of the file, and the names from how I think `connected` was named - I'm happy to adjust though.

Reply:
Oh actually I should have said `is_path_connected`, I'll amend in my next commit